### DISCUSSION TITLE
context variables documentation

### DISCUSSION THREAD

**Discussion by anonkwe** [inventree/InvenTree#5035] [Q&A] [answered]
is there any glossary or list of contexts that we can cast on labels? I'm trying to get the supplier of the part attached in a stockitem label but i'm keep on trying and not finding. is there any doc or does anyone knows the anser of my question?+

tried {{ part.supplier }} , {{ supplier }}, {{ item.supplier }} but none of them are working for now.

any idea?

**Comment by SchrodingersGat** [comment]
@anonkwe a good start would be the docs - https://docs.inventree.org/en/latest/report/labels/#stock-item-labels

**Reply by anonkwe** [reply]
@SchrodingersGat already checked. it doesnt mention it

**Reply by SchrodingersGat** [reply]
@anonkwe try `supplier_part` e.g. `{{ item.supplier_part.SKU }}`

Let me know how you go, sounds like we need to update the docs

**Reply by anonkwe** [reply]
still looking for the context variable to show the part supplier on the stock item label. cant find it anywhere, tried a lot of things.. @SchrodingersGat  do you know it ?

**Reply by miggland** [reply]
Have you tried `{{ item.supplier_part.supplier }}`  or `{{ item.supplier_part.supplier.name }}`?

**Reply by wolflu05** [reply]
If nothing helps, start InvenTree in a devcontainer (you can also use codespaces for that), and use the debugging feature. Set a breakpoint in your template and inspect the context in vscode if you load that template.

**Comment by anonkwe** [comment, marked as answer]
thanks everyone 
{{ item.supplier_part.supplier.name }} worked like a charm!

Just to let you know,  {{ item.supplier_part.supplier }} works as well but it comes out with a hyphen at the end of the supplier name.
